Publish SNAPSHOTs

We should publish snapshots for each successful build of master branch on travis-ci so people can try the lastest code without building it from source.

I am able to publish snapshot for `jetcd`, `jetcd-core`, and `jetcd-discovery-dns-srv` but not `jetcd-watch-example`.
```sh
$mvn deploy -DskipTests=true
...
INFO] --- maven-deploy-plugin:2.7:deploy (default-deploy) @ jetcd-watch-example ---
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO]
[INFO] jetcd .............................................. SUCCESS [  3.345 s]
[INFO] jetcd-core ......................................... SUCCESS [ 26.028 s]
[INFO] jetcd-discovery-dns-srv ............................ SUCCESS [  3.483 s]
[INFO] jetcd-watch-example ................................ FAILURE [  0.370 s]
[INFO] jetcd-simple-ctl ................................... SKIPPED
[INFO] jetcd-examples ..................................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 34.170 s
[INFO] Finished at: 2018-02-13T15:34:51-08:00
[INFO] Final Memory: 91M/356M
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-deploy-plugin:2.7:deploy (default-deploy) on project jetcd-watch-example: Deployment failed: repository element was not specified in the POM inside distributionManagement element or in -DaltDeploymentRepository=id::layout::url parameter -> [Help 1]
```

Investigating on how to skip `jetcd-watch-example`
cc/ @lburgazzoli any idea?

--- a/.travis.yml
+++ b/.travis.yml
@@ -18,3 +18,12 @@ script:
   - sdk use java 8.0.172-zulu
   - java -version
   - ./mvnw clean install
+
+deploy:
+  provider: script
+  script: ./mvnw deploy --settings etc/mvn-deploy-settings.xml
+  skip_cleanup: true 
+  on:
+    branch: release
+    condition: $TRAVIS_PULL_REQUEST = "false"
+    

--- a/etc/mvn-deploy-settings.xml
+++ b/etc/mvn-deploy-settings.xml
@@ -1,0 +1,35 @@
+<?xml version="1.0" encoding="UTF-8"?>
+
+<!--
+
+    Copyright 2017 The jetcd authors
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->
+
+<settings xmlns="http://maven.apache.org/SETTINGS/1.0.0"
+          xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+          xsi:schemaLocation="
+            http://maven.apache.org/SETTINGS/1.0.0 
+            http://maven.apache.org/xsd/settings-1.0.0.xsd">
+
+  <servers>
+    <server>
+      <id>ossrh</id>
+      <username>${env.OSSRH_USERNAME}</username>
+      <password>${env.OSSRH_PASSWORD}</password>
+    </server>
+  </servers>
+  
+</settings>

